src/operator/split.rs

```diff
@@ -0,0 +1,138 @@
+                    token = token[start.len()..token.len() - end.len()].to_string();
+                    trimmed = true;
+                }
+            }
+        }
+
+        map(token)
+    };
+
+    while index < input.len() {
+        let mut found_separator = false;
+        let remaining = &input[index..];
+        if remaining.starts_with(separator) {
+            found_separator = true;
+        } else {
+            let mut found = false;
+            for (start, end) in closures {
+                if !remaining.starts_with(start) {
+                    continue;
+                }
+
+                if let Some(end_index) = remaining[start.len()..].find(end) {
+                    index += start.len() + end_index + end.len();
+                    found = true;
+                    break;
+                }
+            }
+
+            if !found {
+                index += 1;
+            }
+        }
+
+        if found_separator {
+            if index != prev_index {
+                tokens.push(transform(input[prev_index..index].to_string()));
+            }
+            index += separator.len();
+            prev_index = index;
+        }
+    }
+
+    if index != prev_index {
+        tokens.push(transform(input[prev_index..index].to_string()));
+    }
+
+    tokens
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn split_with_separator_and_closures_string(
+        input: &str,
+        separator: &str,
+        closures: &HashMap<&'static str, &'static str>,
+    ) -> Vec<String> {
+        split_with_separator_and_closures(input, separator, closures, |s| s)
+    }
+
+    #[test]
+    fn split_works() {
+        assert_eq!(
+            split_with_separator_and_closures_string("power hello", " ", &DEFAULT_CLOSURES),
+            vec!["power", "hello"],
+        );
+        assert_eq!(
+            split_with_separator_and_closures_string("morecomplicated", "ecomp", &DEFAULT_CLOSURES),
+            vec!["mor", "licated"],
+        );
+        assert_eq!(
+            split_with_separator_and_closures_string("owmmowmow", "ow", &DEFAULT_CLOSURES),
+            vec!["mm", "m"],
+        );
+    }
+
+    #[test]
+    fn split_with_closures_works() {
+        assert_eq!(
+            split_with_separator_and_closures_string(
+                "power hello \"good bye\"",
+                " ",
+                &DEFAULT_CLOSURES
+            ),
+            vec!["power", "hello", "good bye"],
+        );
+        assert_eq!(
+            split_with_separator_and_closures_string(
+                "more'ecomp'licated",
+                "ecomp",
+                &DEFAULT_CLOSURES
+            ),
+            vec!["more'ecomp'licated"],
+        );
+        assert_eq!(
+            split_with_separator_and_closures_string(
+                "ow\"mm\"ow'\"mow\"'",
+                "ow",
+                &DEFAULT_CLOSURES
+            ),
+            vec!["mm", "mow"],
```

Comment:
I think this should be `"mm", "\"mow\"` -- I don't think rolling out _every_ layer of quotes is a reasonable thing to do.

We could also consider just leaving them in, tbh

Reply:
Yeah, I'm not set on any way here (not unrolling, unrolling), but leaning towards unrolling since the value is displayed for the user and removing the quotes is less noisy (and mimics the logic of bash evaluation?).

I can change the logic to unroll only the outer layer, unrolling all made more sense when I was writing it